src/app/admin-panel/schedule/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,122 @@
+            } catch {
+              this.tableData.push(new Array(6).fill(' '));
+              this.tableData[j][i] = dailySchedul.value[j];
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /** Method generates pdf-document from form data */
+  onDownload() {
+    const paddingLeft = 24;
+    const lineHeight = 20;
+    const cellHeight = 30;
+    const columnsCount = 6;
+    let lineNumber = 2;
+
+    const pdfDoc = new jsPDF('l', 'pt', 'a4');
+    pdfDoc.addFileToVFS('Roboto-Regular-normal.ttf', robotoFont);
+    pdfDoc.addFont('Roboto-Regular-normal.ttf', 'Roboto-Regular', 'normal');
+    pdfDoc.setFont('Roboto-Regular');
+
+    // generetes title
+    pdfDoc.setFontSize(14);
+    const title = `Розклад уроків ${this.data.selectedClass} класу`;
+    let xOffset = (pdfDoc.internal.pageSize.width / 2) - (pdfDoc.getStringUnitWidth(title) * pdfDoc.internal.getFontSize() / 2);
+    pdfDoc.text(title, xOffset, lineHeight * lineNumber);
+
+    pdfDoc.setFontSize(12);
+    const period = `на період з ${this.data.dateStart} по ${this.data.dateEnd}`;
+    xOffset = (pdfDoc.internal.pageSize.width / 2) - (pdfDoc.getStringUnitWidth(period) * pdfDoc.internal.getFontSize() / 2);
+    lineNumber += 1;
+    pdfDoc.text(period, xOffset, lineHeight * lineNumber);
+
+    // generetes table headers
+    lineNumber += 2;
+    const firstCellWidth = 20;
+    const cellWidth = (pdfDoc.internal.pageSize.width - paddingLeft * 2 - firstCellWidth) / columnsCount;
+    pdfDoc.cell(paddingLeft, lineHeight * lineNumber, firstCellWidth, lineHeight,
+      ' ', 0, 'center');
+    for (const dayName of this.data.weekDayName) {
+      pdfDoc.cell(paddingLeft, lineHeight * lineNumber, cellWidth, lineHeight,
+        dayName.legendDay, 0, 'center');
+    }
+
+    // generetes table data
+    pdfDoc.setFontSize(10);
+    lineNumber += 1;
+    let row = 1;
+    for (const lesson of this.tableData) {
+      pdfDoc.cell(paddingLeft, lineHeight * lineNumber, firstCellWidth, cellHeight,
+        String(row), row, 'center');
+      for (const subject of lesson) {
+        let cellText = '';
+        if (subject === ' ') {
+          cellText = subject;
+        } else {
+          if ('secondGroup' in subject) {
+            cellText = '';
+            if (subject.firstGroup !== '') {
+              cellText = subject.firstGroup.subjectName;
+            }
+            if (subject.secondGroup !== '') {
+              cellText =  cellText + '\r\n' + subject.secondGroup.subjectName;
+            }
+          } else {
+            cellText = subject.firstGroup.subjectName;
+          }
+        }
+        pdfDoc.cell(paddingLeft, lineHeight * lineNumber, cellWidth, cellHeight,
+          cellText, row, 'center');
+      }
+      row += 1;
+      lineNumber += 1;
+    }
+
+    const fileName = `schedule${this.data.selectedClass}.pdf`;
+    pdfDoc.save(fileName);
+  }
+
+  /** Method closes preview window */
+  onNoClick(): void {
```

Comment:
so if the method closes the preview window, why don't you name it as closePreviewWindow? please use meaningful names